Iterable-iOS-SDKTests/IterableAPIResponseTests.m

```diff
@@ -0,0 +1,160 @@
+//
+//  IterableAPIResponseTests.m
+//  Iterable-iOS-SDKTests
+//
+//  Created by Victor Babenko on 3/29/18.
+//  Copyright © 2018 Iterable. All rights reserved.
+//
+
+#import <XCTest/XCTest.h>
+
+#import "IterableAPI.h"
+#import "OHHTTPStubs.h"
+#import "OHPathHelpers.h"
+
+@interface IterableAPI (ResponseTest)
+- (NSURLRequest *)createRequestForAction:(NSString *)action withArgs:(NSDictionary *)args;
+- (void)sendRequest:(NSURLRequest *)request onSuccess:(void (^)(NSDictionary *))onSuccess onFailure:(void (^)(NSString *, NSData *))onFailure;
+@end
+
+@interface IterableAPIResponseTests : XCTestCase
+
+@end
+
+@implementation IterableAPIResponseTests
+
+- (void)setUp {
+    [super setUp];
+    [IterableAPI sharedInstanceWithApiKey:@"" andEmail:@"" launchOptions:nil];
+}
+
+- (void)tearDown {
+    [OHHTTPStubs removeAllStubs];
+    [super tearDown];
+}
+
+- (NSDictionary *)defaultResponseHeaders {
+    return @{@"Content-Type":@"application/json"};
+}
+
+- (void)stubAnyRequestReturningStatusCode:(int)statusCode data:(NSData *)data {
+    [OHHTTPStubs stubRequestsPassingTest:^BOOL(NSURLRequest * _Nonnull request) {
+        return YES;
+    } withStubResponse:^OHHTTPStubsResponse * _Nonnull(NSURLRequest * _Nonnull request) {
+        return [OHHTTPStubsResponse responseWithData:data statusCode:statusCode headers:self.defaultResponseHeaders];
+    }];
+}
+
+- (void)stubAnyRequestReturningStatusCode:(int)statusCode json:(NSDictionary *)json {
+    NSError *error = nil;
+    NSData *jsonResponseData = [NSJSONSerialization dataWithJSONObject:json
+                                                               options:0 error:&error];
+    [self stubAnyRequestReturningStatusCode:statusCode data:jsonResponseData];
+}
+
+- (void)testResponseCode200 {
+    NSDictionary *responseData = @{@"key":@"value"};
+    [self stubAnyRequestReturningStatusCode:200 json:responseData];
+    
+    XCTestExpectation *expectation = [[XCTestExpectation alloc] initWithDescription:@"onSuccess is called"];
+    
+    NSURLRequest *request = [[IterableAPI sharedInstance] createRequestForAction:@"" withArgs:@{}];
+    [[IterableAPI sharedInstance] sendRequest:request onSuccess:^(NSDictionary * _Nonnull data) {
+        [expectation fulfill];
+        XCTAssert([data isEqualToDictionary:responseData]);
+    } onFailure:nil];
+    [self waitForExpectations:@[expectation] timeout:1.0];
```

Comment:
Since it is a stubbed test, how did you choose a timeout of 1.0?

Reply:
We're testing whether onSuccess is called. If it's not, we want to fail after some reasonable timeout.

Reply:
Can you change it to be a global constant since you use the same timeout?

Reply:
Done